components/core/src/clp/clp/FileDecompressor.cpp

```diff
@@ -1,13 +1,55 @@
 #include "FileDecompressor.hpp"
 
+#include <boost/filesystem.hpp>
 #include <boost/filesystem/operations.hpp>
 #include <boost/filesystem/path.hpp>
 
-#include "../spdlog_with_specializations.hpp"
+#include "../ir/LogEventSerializer.hpp"
+#include "../ir/utils.hpp"
 
+using clp::ir::four_byte_encoded_variable_t;
+using clp::ir::LogEventSerializer;
 using std::string;
 
 namespace clp::clp {
+namespace {
+/**
+ * Rename a temporary IR and move it to the output directory
+ * The new name follows the following format
+ * <FILE_ORIG_ID>_<begin_message_ix>_<end_message_ix>.clp.zst
+ * @param temp_ir
+ * @param output_directory
+ * @param file_orig_id
+ * @param begin_message_ix
+ * @param end_message_ix
+ * @return true if the IR is renamed and moved successfully. Otherwise false
+ */
+bool rename_and_move_ir(
```

Comment:
"rename" and "move" here should have the same meaning. We should call it either "rename_ir_file" or "move_ir_file"